docusaurus/docs/React/custom-code-examples/channel-search.mdx

```diff
@@ -0,0 +1,305 @@
+---
+id: channel_search
+sidebar_position: 4
+title: ChannelSearch
+---
+
+import CustomChannelSearch from '../assets/CustomChannelSearch.png';
+
+## How-to Guide for Customizing ChannelSearch
+
+In this example we will customize the [`ChannelSearch`](../core-components/channel_list/#channelsearch) component. Though this component can be used standalone, we will do our customization when it's established in the `ChannelList`. Here is a guide that outlines how to
+override the default `DropdownContainer` and `SearchResultItem` components, as well as use several other useful props.
+
+:::note
+A complete override of this component is possible by utilizing the [`ChannelSearch`](../core-components/channel_list/#channelsearch) prop on `ChannelList`.
+:::
+
+### Step 1 - Setting Up For Success
+
+To make the search component visible in the `ChannelList`, you need to use the `showChannelSearch` prop. At this point, the search component is ready to go with all of the default UI and functionality.
+
+Let's allow the query to search for `channels` in addition to the default query of just users by setting the `searchForChannels` prop. You'll notice that all props for `ChannelSearch`
+are drilled to the component via the `additionalChannelSearchProps` prop on `ChannelList`. With just this one additional prop, the search functionality is fully functional and expanded from just the default search of users.
+
+```jsx
+<ChannelList
+    showChannelSearch
+    additionalChannelSearchProps={
+        {
+          searchForChannels: true,
+        }
+    }
+/>
+```
+
+### Step 2 - Customizing the DropdownContainer
+
+Since we've enabled the search to also search for channels, let's customize the dropdown results container to fit a design that separates the result by type.
+
+```jsx
+  const CustomDropdown = (props: DropdownContainerProps) => {
+    const { results, focusedUser, selectResult, SearchResultItem } = props;
+
+    let items: ChannelOrUserType[] = results.filter(x => x.cid);
+    let users: ChannelOrUserType[] = results.filter(x => !x.cid);
+
+    return (
+      <div>
+        <p>Channels</p>
+        {items.map((result: ChannelOrUserType, index: number) => (
+          <SearchResultItem
+            focusedUser={focusedUser}
+            index={index}
+            key={index}
+            result={result}
+            selectResult={selectResult}
+          />
+        ))}
+        <p>Users</p>
+        {users.map((result: ChannelOrUserType, index: number) => (
+          <SearchResultItem
+            focusedUser={focusedUser}
+            index={index}
+            key={index}
+            result={result}
+            selectResult={selectResult}
+          />
+        ))}
+      </div>
+    );
+  };
+```
+
+```jsx
+<ChannelList
+    showChannelSearch
+    additionalChannelSearchProps={
+        {
+            searchForChannels: true,
+            DropdownContainer: (props: DropdownContainerProps) => <CustomDropdown {...props} />
+        }
```

Comment:
again looks a little odd like this, maybe declare a const outside the jsx and pass to the prop?